subsys/bluetooth/audio/unicast_client.c

```diff
@@ -366,60 +357,76 @@ static struct bt_audio_ep *unicast_client_ep_find(struct bt_conn *conn,
 	return NULL;
 }
 
+struct audio_iso_find_params {
+	enum bt_audio_dir dir;
+	const struct bt_audio_unicast_group *unicast_group;
+	struct bt_conn *acl;
+};
+
+static bool audio_iso_find_unused_cb(struct bt_audio_iso *audio_iso, void *user_data)
+{
+	struct audio_iso_find_params *params = user_data;
+	const struct bt_audio_stream *stream, *paired_stream;
+
+	if (params->dir == BT_AUDIO_DIR_SINK) {
+		stream = audio_iso->unicast.sink_stream;
+		paired_stream = audio_iso->unicast.source_stream;
+	} else {
+		stream = audio_iso->unicast.source_stream;
+		paired_stream = audio_iso->unicast.sink_stream;
+	}
+
+	if (stream != NULL || paired_stream == NULL) {
+		return false;
+	}
```

Comment:
Not sure I follow why we are returning `false` if `paired_stream == NULL`. Can you elaborate?